### ISSUE TITLE
VIA 8237

### ISSUE THREAD

**Issue by mbodomi** [crazii/SBEMU#7]
Confirmed as not working

**Comment by msantos2007** [comment]
Working fine on VIA VT8237A.

More details:
- SBEMU 1.0 Beta 3 (Vogons)
- Biostar P4M900 M7 Motherboard *
- MS DOS 6.22
- EMM386 NOEMS (jemmex/jload not used, but also works)

* Also worked on Intel DG41RQ/DG41TY

**Comment by msantos2007** [comment]
Working fine on VIA VT8237A.


More details:

SBEMU 1.0 Beta 3 (Vogons)
Biostar P4M900 M7 Motherboard *
MS DOS 6.22
EMM386 NOEMS (jemmex/jload not used, but also works)
Doom 1.9


*Also worked on Intel DG41RQ/DG41TY motherboard

**Comment by msantos2007** [comment]
Working fine on VIA VT8237A.

More details:

- SBEMU 1.0 Beta 3 (Vogons)
- Biostar P4M900 M7 Motherboard *
- MS DOS 6.22
- EMM386 NOEMS (jemmex/jload not used, but works)
- Doom 1.9


     *Also worked on Intel DG41RQ/DG41TY motherboard

**Comment by msantos2007** [comment]
Working fine on VIA VT8237A.

More details:

- SBEMU 1.0 Beta 3 (Vogons)
- Biostar P4M900 M7 Motherboard *
- MS DOS 6.22
- EMM386 NOEMS
- Doom
- jemmex/jload not used, but also works

  *Also worked on Intel DG41RQ/DG41TY motherboard

**Comment by msantos2007** [comment]
Working fine on VIA VT8237A.

More details:

- SBEMU 1.0 Beta 3 (Vogons)
- Biostar P4M900 M7 Motherboard *
- MS DOS 6.22
- EMM386 NOEMS
- Doom
- jemmex/jload not used, but also work
 
 
  
*Also worked on Biostar P4M890 M7 and Intel DG41RQ/DG41TY motherboards

**Comment by Torinde** [comment]
The [VSBHDA fork has some VT82xx improvements](https://github.com/Baron-von-Riedesel/VSBHDA/commits/main) and mentions in its readme VT8233/35/37.

**Comment by crazii** [comment]
> The [VSBHDA fork has some VT82xx improvements](https://github.com/Baron-von-Riedesel/VSBHDA/commits/main) and mentions in its readme VT8233/35/37.

I've checked the code, VSBHDA has the same way fixing for VT8237, and the VT8237 in VSBHDA is inspired from SBEMU, since the beta3 with VT8237 of SBEMU comes first.

**Comment by crazii** [comment]
> OK, so SBEMU already supports VT8233/35/37? Or only VT8233 and VT8237?
> 
> Are there other improvements from VSBHDA that can be merged back into SBEMU (besides the [compatibility with latest upstream HX](https://github.com/crazii/SBEMU/issues/40))?
> 
> Also, VSBHDA mentions that it emulates Sound Blaster Pro2, while SBEMU doesn't mention that? I assume it's just an omission and Pro2 is emulated by SBEMU?

VT8233/35/37 all supported, they're using the same driver, but I only tested 8027.
There might be improvements for HDA in VSBHDA, but I haven't checked. 

I didn't explicitly state sbpro2. pro2 is no difference from pro, except for opl3. And pro2 is also a pro card, I remember Miles Sound have "sbpro (old version)" "sbpro (new version)" for music, but not for digital sound. Anyways it would be nice to add sbpro2 to avoid misunderstanding.

**Comment by crazii** [comment]
Yes not to do that in a easy way. I might be wrong but it seems that he's willingly to share his HX and code, but not willingly to use/help with others' code, based on the fact that he ignored my PR suggestion on port trapping, and that he totally reformed sbemu into another different shape and no PRs back. 
Now sbemu has more clean code in using mpxplay, but he just wouldn't wait. 
I'm glad if someone would create PRs based on his changes, but I'm not gonna do that myself, it's just - weird. It's not how open source work, at least not to me.

**Comment by crazii** [comment]
Yes  thanks for the reminding. I think it's doable, but as you said I need little more time too. Currently I'm debugging usbddos and try to make it work on @volkertb 's PCs. But it seems difficult because I cannot debug, I have to check/review the code and find what the problem is. Maybe to buy a 2nd handed Mac Mini 2011 should be the right solution. Still there's some bugs I've encountered, and I'm debugging it now.
Fortunately I'm not back to working yet, there'll be 1 or 2 months for me, otherwise the time would be more of a problem.
